azure-ts-dynamicresource/cdnCustomDomain.ts

```diff
@@ -0,0 +1,231 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as azure from "@pulumi/azure";
+
+import * as msRestAzure from "@azure/ms-rest-nodeauth";
+import * as cdnManagement from "@azure/arm-cdn";
+
+export interface CustomDomainOptions {
+  resourceGroupName: string;
+  profileName: string;
+  endpointName: string;
+  customDomainName: string;
+  customDomainHostName: string;
+  httpsEnabled: boolean;
+}
+
+export interface CustomDomainOutputs extends cdnManagement.CdnManagementModels.CustomDomainsCreateResponse {
+  inputs: CustomDomainOptions;
+}
+
+class CDNCustomDomainResourceProvider implements pulumi.dynamic.ResourceProvider {
+
+  private async getCDNManagementClient(): Promise<cdnManagement.CdnManagementClient> {
+    let clientID = azure.config.clientId;
+    let clientSecret = azure.config.clientSecret;
+    let tenantID = azure.config.tenantId;
+    let subscriptionID = azure.config.subscriptionId;
+
+    // If at least one of them is empty, try looking at the env vars.
+    if (!clientID || !clientSecret || !tenantID || !subscriptionID) {
+      console.log("Checking env vars for ARM credentials.");
+      clientID = process.env["ARM_CLIENT_ID"];
+      clientSecret = process.env["ARM_CLIENT_SECRET"];
+      tenantID = process.env["ARM_TENANT_ID"];
+      subscriptionID = process.env["ARM_SUBSCRIPTION_ID"];
+    }
+
+    // If they are still empty, then throw an error.
+    if (!clientID || !clientSecret || !tenantID || !subscriptionID) {
+      throw new Error("ARM credentials could not be detected.");
+    }
+
+    const credentials = await msRestAzure.loginWithServicePrincipalSecret(clientID, clientSecret, tenantID);
```

Comment:
This fails when I run it from CLI:

> Checking env vars for ARM credentials.
> Error: ARM credentials could not be detected.: Error: ARM credentials could not be detected.

Shouldn't we do something like `const cred = await msRestAzure.AzureCliCredentials.create();` instead?

Reply:
The dynamic provider tries to use the Pulumi CLI's Azure config, and if not looks for environment vars. I haven't tried using `msRestAzure.AzureCliCredentials.create();` myself, to be honest. I did it this way since I typically use a Service Connection in Azure DevOps and don't use the Az CLI in any of my tasks.

Reply:
The README suggests `az login`, so it should work. Also, that's probably what users will use while trying examples.

Reply:
Good catch! I have updated this to try and get the credentials from the CLI if the env vars are not set.